docs/primitives.md

```diff
@@ -0,0 +1,41 @@
+# Primitives
+
+Caesium's primary function is a job scheduler and executor. As such, the design and implementation of this process will revolve around a set of primitives which aid in defining the lifcycle of a job. Below is an outline of how the primitives will be structured and implemented. Note that this is not a detailed implementation proposal, but rather meant merely as a holistic, pre-implementation guide for how Caesium's internals will be structured.
+
+## `Job`
+
+A `Job` is the primary unit of work in Caesium. A `Job` is analagous to a `DAG` in Airflow, and will also be defined as an acyclic graph. A job is is made up of several other primitives described below.
+
```

Comment:
Would you consider a `Job` to be the primary  unit of work, or a `Task`?

Reply:
It's a fair point - I struggled with the language here. I want to establish that if you want to do something in Caesium, you want to write a `Job` for it, rather than having a single `Job` that someone dumps a bunch of `Task`s in. Do you have an idea for a better way to communicate that?

Reply:
Ah. I see what you're trying to say. 
Maybe something like 
A `Job` is collection of related tasks. All the tasks in a single `Job` will depend on the same `Trigger`(s), or the completion of the other `Task`s in the `Job`. A job is analogous to a `DAG` in Airflow., and is also defined as a directed acyclic graph. A `Job` can container other primitive types in addition to `Task`s, as described below.

Reply:
Feedback applied, and also added more source files for the other primitives.